Sources/Dependencies/DependencyValues.swift

```diff
@@ -477,7 +477,12 @@ public final class CachedValues: @unchecked Sendable {
     return withIssueContext(fileID: fileID, filePath: filePath, line: line, column: column) {
       let cacheKey = CacheKey(id: TypeIdentifier(key), context: context)
       #if DEBUG
-        if context == .live, !DependencyValues.isSetting, !(key is any DependencyKey.Type) {
+        if
+          context == .live,
+          !DependencyValues.isSetting,
+          !(cached[cacheKey] != nil && cached[cacheKey]?.preparationID != nil),
```

Comment:
We do not need to warn if there is something in the cache that came from preparing dependencies.